frontend/src/plugin/registry.tsx

```diff
@@ -758,6 +771,147 @@ export function registerKindIcon(kind: string, definition: IconDefinition) {
   store.dispatch(graphViewSlice.actions.addKindIcon({ kind, definition }));
 }
 
+/**
+ * Register a new cluster action menu item.
+ * @param item - The item to add to the cluster action menu.
+ *
+ * @example
+ *
+ * ```tsx
+ * import { registerClusterProviderMenuItem } from '@kinvolk/headlamp-plugin/lib';
+ * import { MenuItem, ListItemText } from '@mui/material';
+ * registerClusterProviderMenuItem(({cluster, setOpenConfirmDialog, handleMenuClose}) => {
+ *  const isMinikube =
+ *   cluster.meta_data?.extensions?.context_info?.provider === 'minikube.sigs.k8s.io';
+ *   if (!helpers.isElectron() !! !isMinikube) {
+ *     return null;
+ *   }
+ *   return (
+ *     <MenuItem
+ *       onClick={() => {
+ *        setOpenConfirmDialog('deleteMinikube');
+ *        handleMenuClose();
+ *       }}
+ *     >
+ *       <ListItemText>{t('translation|Delete')}</ListItemText>
+ *     </MenuItem>
+ *   );
+ * )}
+ * ```
+ *
+ */
+export function registerClusterProviderMenuItem(item: MenuItemComponent) {
+  store.dispatch(addMenuItem(item));
+}
+
+/**
+ * Register a new cluster provider dialog.
+ * @param item - The item to add to the cluster action dialog.
+ *
+ * @example
+ *
+ * ```tsx
+ * import { registerClusterProviderDialog } from '@kinvolk/headlamp-plugin/lib';
+ * import { CommandCluster } from './CommandCluster';
+ *
+ * registerClusterProviderDialog(({cluster, active, openConfirmDialog, setOpenConfirmDialog}) => {
+ *
+ *   const isMinikube =
+ *   cluster.meta_data?.extensions?.context_info?.provider === 'minikube.sigs.k8s.io';
+ *   if (!helpers.isElectron() !! !isMinikube) {
+ *     return null;
+ *   }
+ *
+ *   return (
+ *     <CommandCluster
+ *       initialClusterName={cluster.name}
+ *       open={openConfirmDialog === 'startMinikube'}
+ *       handleClose={() => setOpenConfirmDialog('')}
+ *       onConfirm={() => {
+ *         setOpenConfirmDialog('');
+ *       }}
+ *       command={'start'}
+ *       finishedText={'Done! kubectl is now configured'}
+ *     />
+ *   );
+ * });
+ *
+ * ``
+ */
+export function registerClusterProviderDialog(item: DialogComponent) {
+  store.dispatch(addDialog(item));
+}
+
+/**
+ * For adding a card to the Add Cluster page in the providers list.
```

Comment:
this seems to be too permissive, as it allows plugins to render pretty much anything here which will result in an inconsistent UI depending on a plugin

would an interface with {icon, description, url} definition be enough for this part?

Reply:
done